public/api/supportFileUpload.php

```diff
@@ -0,0 +1,168 @@
+<?php
+header('Access-Control-Allow-Origin: *');
+header('Access-Control-Allow-Headers: access');
+header('Access-Control-Allow-Methods: POST');
+header('Access-Control-Allow-Credentials: true');
+header('Content-Type: application/json');
+
+include 'db_connection.php';
+$jwtArray = include 'jwtArray.php';
+$userId = $jwtArray['userId'];
+
+// include "randomId.php";
+include 'userQuotaBytesAvailable.php';
+include 'getFilename.php';
+include 'cidFromSHA.php';
+
+$doenetId = mysqli_real_escape_string($conn, $_POST['doenetId']);
+
+$success = true;
+$msg = '';
+
+$uploads_dir = '../media/';
+
+$type = $_FILES['file']['type'];
+$tmp_name = $_FILES['file']['tmp_name'];
+$error = $_FILES['file']['error'];
+$size = $_FILES['file']['size'];
+$original_file_name = $_FILES['file']['name'];
+$description = ''; //Don't automatically fill with file name
+// $description = substr($original_file_name, 0, strrpos($original_file_name, "."));
+
+$tmp_dest = $uploads_dir . getFileName('tmp_' . $random_id, $type);
+
+if ($type != "image/png" && $type != "image/jpeg"){
+    $success = false;
+    $msg = "Only .png or .jpg files. '$original_file_name' can not be uploaded.";
+}
+//Test if user has space to upload files
+if ($success) {
```

Comment:
There is a lot of shared code with the thumbnail API, I know that has the additional restriction of only one per doenetID, but it would be good to figure out how to share the common code between the two.